add option to allow processing of unexported types

ebfe opened the pull request:
<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/tinylib/msgp/95)
<!-- Reviewable:end -->

ebfe commented:
Consider this an RFC. Things are named pretty awkwardly. Suggestions welcome.

ttacon commented on main.go:
```diff
@@ -41,6 +41,7 @@ var (
 	encode  = flag.Bool("io", true, "create Encode and Decode methods")
 	marshal = flag.Bool("marshal", true, "create Marshal and Unmarshal methods")
 	tests   = flag.Bool("tests", true, "create tests and benchmarks")
+	exports = flag.Bool("exports", true, "process only exported types")
```
Personally, I'd rather this be the other way. Otherwise usage is a little awkward. For example:

```
msgp -exports=false ...
```

is far less useable/readable as opposed to if you defined the flag to be something along the lines:

```go
var allTypes = flag.Bool("all-types", false, "process all types")
```

which means usage would be

```
msgp -all-types ...
```

which is far more readable IMHO, but that's just my two cents.

philhofer replied:
I agree. `unexported` also works.

ttacon commented on parse/getast.go:
```diff
@@ -23,10 +24,11 @@ type FileSet struct {
 // File parses a file at the relative path
 // provided and produces a new *FileSet.
-// (No exported structs is considered an error.)
+// (No structs is considered an error.)
```
Doesn't this maintain the previous behavior if the flag to process local imports isn't provided? The documentation should make this clear, currently it makes me think that this would change the previous behaviour.

philhofer replied:
It shouldn't say "structs," either, since we've been able to process most composite types for a while now.

philhofer commented:
Sorry it took me a while to find time to look at this.

If `unexported` (or whatever) is a global processing flag, then it doesn't need to be in `modes.` (Those are for generation paths that can be turned on/off per type declaration.)

ebfe commented:
Updated. I think I addressed most (all?) of the comments.

Unfortunately another problem showed up. The generated benchmarks/tests for
unexported types are not picked up by `go test`.

ebfe commented:
I think the problem with benchmarks/tests is that the function names now begin with Test[lowercase] while `go test` only looks for Test[uppercase] prefixes.

philhofer commented:
I think the solution is to invert the naming of the existing tests. Instead of `Test[name]Unmarshal`, we should generate `TestUnmarshal[name]`.

ebfe commented:
Done

philhofer commented:
Great. LGTM.
